src/server/endpoints.rs

```diff
@@ -119,49 +180,48 @@ pub(crate) async fn print_file(
     let ctx = rqctx.context().clone();
     let machine_id = params.machine_id.clone();
     let job_id = uuid::Uuid::new_v4();
+    let job_name = &params.job_name;
+
+    let machines = ctx.machines.read().await;
+    let machine = match machines.get(&machine_id) {
+        Some(machine) => machine,
+        None => {
+            tracing::warn!(id = machine_id, "machine not found");
+            return Err(HttpError::for_not_found(
+                None,
+                format!("machine not found by id: {:?}", machine_id),
+            ));
+        }
+    };
+
+    let filepath = std::env::temp_dir().join(format!(
+        "{}_{}",
+        job_id.simple(),
+        file.file_name.unwrap_or("file".to_string())
+    ));
+    tracing::info!(path = format!("{:?}", filepath), "Writing file to disk");
 
-    let machine = ctx
-        .find_machine_handle_by_id(&machine_id)
-        .map_err(|e| {
-            tracing::error!("failed to find machine by id: {:?}", e);
-            HttpError::for_bad_request(None, format!("machine not found by id: {:?}", machine_id))
-        })?
-        .ok_or_else(|| {
-            tracing::error!("machine not found by id: {:?}", machine_id);
-            HttpError::for_not_found(None, format!("machine not found by id: {:?}", machine_id))
-        })?;
-    let filepath = std::env::temp_dir().join(format!("{}-{}", job_id, file.file_name.unwrap_or("file".to_string())));
     // TODO: we likely want to use the kittycad api to convert the file to the right format if its
     // not already an stl file.
+
     tokio::fs::write(&filepath, file.content).await.map_err(|e| {
-        tracing::error!("failed to write stl file: {:?}", e);
+        tracing::error!(error = format!("{:?}", e), "failed to write stl file");
         HttpError::for_bad_request(None, "failed to write stl file".to_string())
     })?;
 
-    match machine {
-        crate::machine::MachineHandle::UsbPrinter(_) => {
-            let print_job = PrintJob {
-                file: filepath,
-                machine,
-                job_name: params.job_name.to_string(),
-            };
-            let handle = print_job.spawn().await;
-            let mut active_jobs = ctx.active_jobs.lock().await;
-            active_jobs.insert(job_id.to_string(), handle);
-        }
-        crate::machine::MachineHandle::NetworkPrinter(printer) => {
-            let result = printer
```

Comment:
just a dumb thing but i just wanna make sure for network printers its not queued up somewhere since like it can be sliced and sent to the printer in the request, and i say this only because sometimes the slice will fail, because like the print is too high for the printer or something and i like getting those errors direct. didnt look too far at code so you might already be doing this but just a heads up, also i made everything external errors since like its running on your network and you likely wanna know what the eff happened versus it being invisible to you

Reply:
@jessfraz word totally agree -- this specific code from [main](https://github.com/KittyCAD/machine-api/blob/452ed9b569cc575099ed88034ffcd775ea892a69/src/server/endpoints.rs#L142-L151) -- i was hoping to start to chip away on things like the actual queue/jobs where we can push everything through (state, errors, etc) -- but I totally agree -- mind if that's a different pr/line of work? I feel bad this PR is this huge already

Reply:
@jessfraz yeah word -- when you hit `/print`, the [endpoint](https://github.com/KittyCAD/machine-api/blob/paultag/refactor/src/server/endpoints.rs#L212-L230) will call into `.build`, which [will slice on the request thread](https://github.com/KittyCAD/machine-api/blob/paultag/refactor/src/machine.rs#L47-L68) -- then send that to the printer's control channel implementation -- which is machine-specific, but we could enforce some sort of policy for how much gets done before it returns a success/failure. 

We get the error from the [slicer](https://github.com/KittyCAD/machine-api/blob/paultag/refactor/src/slicer/prusa.rs#L88) so I think we should be able to make sure it gets returned to the user, although I'm not sure it does right now. I'll look at this again tomorrow with fresh brains